JavaFiddle-ejb/src/test/java/GroupBeanTest.java

```diff
@@ -0,0 +1,163 @@
+        Path destinationPath = Paths.get(System.getProperty("user.dir")+"/src/main/resources/META-INF/persistence.xml");
+        Path destinationPath2 = Paths.get(System.getProperty("user.dir")+"/target/classes//META-INF/persistence.xml");
+        java.io.File temp = new java.io.File(System.getProperty("user.dir")+"/src/main/resources/META-INF/persistence.xml");
+        java.io.File temp2 = new java.io.File(System.getProperty("user.dir")+"/target/classes/META-INF/persistence.xml");
+        if(temp.exists() || temp2.exists()) {
+            temp.delete();
+            temp2.delete();
+        }
+        Files.copy(sourcePath, destinationPath);
+        Files.copy(sourcePath, destinationPath2);
+        System.out.println("Test EJBContainer is closed" );
+    }
+//Some small functions are used in these three or four main ones.
+
+    @Test
+    public void testGroupOperations() throws UnsupportedEncodingException, NoSuchAlgorithmException, InstantiationException, IllegalAccessException {
+
+        GroupBean groupBean = null;
+        ProjectBean projectBean = null;
+        UserBean userBean = null;
+        AccessBean accessBean = null;
+        UserGroupBean userGroupBean = null;
+
+        try {
+            groupBean = (GroupBean) context.lookup("java:global/classes/GroupBean");
+            projectBean = (ProjectBean) context.lookup("java:global/classes/ProjectBean");
+            userBean = (UserBean) context.lookup("java:global/classes/UserBean");
+            accessBean = (AccessBean) context.lookup("java:global/classes/AccessBean");
+            userGroupBean = (UserGroupBean) context.lookup("java:global/classes/UserGroupBean");
+        } catch (NamingException ex) {
+            System.out.println("Unable to initialize UserBean instance: " + ex);
+        }
+
+
+
+        initializeAndCreate(groupBean, projectBean,userBean, accessBean);
+        int groupId = groupBean.getGroupByGroupId(1).getGroupId();
+      //  userGroupBean.getUserGroup(1,1);
+//added member-----------------------------------------------------------
+        Group group = groupBean.getGroupByGroupId(1);
+        User user1 = userBean.getUser("barny");
+        User user2 = userBean.getUser("uollis");
+        Access access1 = accessBean.getAccess("partial");
+        Access access2 = accessBean.getAccess("full");
+        groupBean.addMember(group,user1,access1);
+        groupBean.addMember(group,user2,access2);
+//check if member was added----------------------------------------------
+        group = groupBean.getGroupByGroupId(1);
+        Assert.assertNotNull(group);
+        System.out.println(group.getGroupName());
+        List<UserGroup> gr = groupBean.getGroupMembers(1);
+
+        Assert.assertNotNull("GET MEMBERS RETURNED NULL",gr);
+        Assert.assertFalse("THE LIST OF MEMBERS IS EMPTY", gr.isEmpty());
+        if (gr == null || gr.isEmpty()) {
+
+            logger.log(Level.SEVERE, "NO ENTITY WAS ADDED");
+            System.exit(-1);
```

Comment:
Wow-wow. You're forcing your java process exit. Imagine this running on Jenkins.
It's an overkill. You can simply fail your particular test using Assert.fail(String message)